src/containers/Movie/MovieInfo/Actions/container.js

```diff
@@ -0,0 +1,77 @@
+import React, { Component } from 'react'
+import { connect } from 'react-redux'
+import PropTypes from 'prop-types'
+
+import {
+  getMovieAccountStates as getMovieAccountStatesAction,
+  toggleMovieWatchlist as toggleMovieWatchlistAction,
+  addMovieToFavorites as addMovieToFavoritesAction
+} from 'Store/movie/actions'
+import {
+  accountStatesLoadingSelector,
+  isOnWatchlistSelector,
+  isFavoriteSelector,
+  currentMovieIdSelector
+} from 'Store/movie/selectors'
+import ActionsComponent from './component'
+
+class Actions extends Component {
+  componentDidMount() {
+    const { getMovieAccountStates, movieId } = this.props
+    getMovieAccountStates(movieId)
+  }
+
+  handleToggleWatchlist = () => {
+    const { movieId, isOnWatchlist, toggleMovieWatchlist } = this.props
+    toggleMovieWatchlist(movieId, isOnWatchlist)
+  }
+
+  handleToggleBookmark = () => {
+    const { movieId, isFavorite, addMovieToFavorites } = this.props
+    addMovieToFavorites(movieId, isFavorite)
+  }
+
+  render() {
+    const { loading, isOnWatchlist, isFavorite } = this.props
+    return (
+      <ActionsComponent
+        loading={loading}
+        isOnWatchlist={isOnWatchlist}
+        isFavorite={isFavorite}
+        handleToggleWatchlist={this.handleToggleWatchlist}
+        handleToggleBookmark={this.handleToggleBookmark}
+      />
+    )
+  }
+}
+
+Actions.propTypes = {
+  loading: PropTypes.bool.isRequired,
+  getMovieAccountStates: PropTypes.func.isRequired,
+  movieId: PropTypes.string,
+  isOnWatchlist: PropTypes.bool,
+  toggleMovieWatchlist: PropTypes.func.isRequired,
+  isFavorite: PropTypes.bool,
+  addMovieToFavorites: PropTypes.func.isRequired
+}
+
+Actions.defaultProps = {
+  movieId: null,
+  isOnWatchlist: null,
+  isFavorite: null
+}
+
+const mapStateToProps = (state, { movieId }) => ({
+  loading: accountStatesLoadingSelector(state),
+  isOnWatchlist: isOnWatchlistSelector(state, movieId),
+  isFavorite: isFavoriteSelector(state, movieId),
+  movieId: currentMovieIdSelector(state)
```

Comment:
you already have movieId from props

Reply:
@VladWithCookies yes, but not from the parent component but from the `connect` function

Reply:
what is the difference?

Reply:
removed `movieId`